### ISSUE TITLE
MSFT_SPSearchContentSource does not take StartHour parameter for schedule

### ISSUE THREAD

**Issue by adruet29** [PowerShell/SharePointDsc#840]
**Details of the scenario you try and problem that is occurring:**

For a Sharepoint source, when setting a StartHour different from 0 in a content source crawl schedule not warning or error is issued, but the start hour remains 0:00.

**The DSC configuration that is using the resource:**

 SPSearchContentSource LocalSharePointSites
            {
                Name                 = "Local SharePoint Sites"
                ServiceAppName       = "Search Service Application"
                ContentSourceType    = "SharePoint"
                Addresses            = @("http://sharepointsite1.contoso.com", "http://sharepointsite2.contoso.com")
                CrawlSetting         = "CrawlEverything"
                ContinuousCrawl      = $false
                IncrementalSchedule  = MSFT_SPSearchCrawlSchedule{
                                        ScheduleType = "Daily"
                                        StartHour = "6"
                                        StartMinute = "0"
                                        CrawlScheduleRepeatDuration = "1440"
                                        CrawlScheduleRepeatInterval = "5"
                                       }
                FullSchedule         = MSFT_SPSearchCrawlSchedule{
                                        ScheduleType = "Weekly"
                                        CrawlScheduleDaysOfWeek = @("Monday", "Wednesday", "Friday")
                                        StartHour = "5"
                                        StartMinute = "0"
                                       }
                Priority             = "Normal"
                Ensure               = "Present"
                PsDscRunAsCredential = $SetupAccount
            }
**Version of the Operating System and PowerShell the DSC Target Node is running:**
OS: Windows Server 2016
PSVersion                      5.1.14393.2363

**Version of SharePoint that is used (e.g. SharePoint 2016):**
Sharepoint 2016

**Version of the DSC module you're using:**
2.3

**Comment by adruet29** [comment]
I was able to solve this issue, but I must admit that I am a newbie regarding the pull request process...

So here it is. From what I was able to understand, the StartHour parameter is correctly checked in the test Get-SPDSCSearchCrawlSchedule.psm1 module.
The issue seems to be in the MSFT_SPSearchContentSource.psm1, as in its Set-TargetResource function there is never the parameter CrawlScheduleStartDateTime which is required by the sharepoint powershell cmdlet Set-SPEnterpriseSearchCrawlContentSource to had a start hour to the schedule.

So here is my (probably dirty) fix:

Between the line 519 and 520 I added the following:
                ```
$propertyTest = Test-SPDSCObjectHasProperty -Object $params.IncrementalSchedule `
                                                            -PropertyName "StartHour"
                if ($propertyTest -eq $true)
                {
                    $incrementalSetArgs.Add("CrawlScheduleStartDateTime", 
                        "$($params.IncrementalSchedule.StartHour):$($params.IncrementalSchedule.StartMinute)")
                }
```

And between the line 595 and 596 I added the following code:
               ```
 $propertyTest = Test-SPDSCObjectHasProperty -Object $params.FullSchedule `
                                                            -PropertyName "StartHour"
                if ($propertyTest -eq $true)
                {
                    $fullSetArgs.Add("CrawlScheduleStartDateTime", 
                        "$($params.FullSchedule.StartHour):$($params.FullSchedule.StartMinute)")
                }
```
This fixed my issue.

**Comment by adruet29** [comment]
Ok, so I read the rules for contributing, and I hope I did not make any mistakes in my pull request:
[](https://github.com/PowerShell/SharePointDsc/pull/841)

**Comment by ykuijs** [comment]
Hi @adruet29, thanks for your work investigating and fixing this issue. I will let the Unit tests run and will review once completed.

**Comment by ykuijs** [comment]
Fixed by PR #842
